support-frontend/app/services/AuthenticationService.scala

```diff
@@ -1,38 +1,69 @@
 package services
 
-import com.gu.identity.cookie.IdentityKeys
-import com.gu.identity.play.AccessCredentials.{Cookies, Token}
-import com.gu.identity.play.AuthenticatedIdUser
-import com.gu.identity.play.AuthenticatedIdUser._
-import play.api.mvc.RequestHeader
+import cats.implicits._
+import com.gu.identity.auth.UserCredentials
+import com.gu.identity.model.User
+import com.gu.identity.play.IdentityPlayAuthService
+import config.Identity
+import org.http4s.Uri
+import play.api.mvc.{Cookie, RequestHeader}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class AuthenticationService(override val identityKeys: IdentityKeys) extends com.gu.identity.play.AuthenticationService {
-  override lazy val authenticatedIdUserProvider: Provider =
-    Cookies.authProvider(identityKeys).withDisplayNameProvider(Token.authProvider(identityKeys, "membership"))
+// The following classes were previously defined in identity-play-auth,
+// but have been removed as part of the changes to that library:
+// authenticating users via a call to identity API; removing periphery functionality.
+// They have been redefined here to reduce diff across PRs,
+// but these classes could get refactored / simplified / removed in subsequent PRs.
+sealed trait AccessCredentials
+object AccessCredentials {
+  case class Cookies(scGuU: String, guU: Option[String] = None) extends AccessCredentials {
+    val cookies: Seq[Cookie] = Seq(
+      Cookie(name = "SC_GU_U", scGuU)
+    ) ++ guU.map(c => Cookie(name = "GU_U", c))
+  }
+  case class Token(tokenText: String) extends AccessCredentials
 }
+case class IdMinimalUser(id: String, displayName: Option[String])
+case class AuthenticatedIdUser(credentials: AccessCredentials, user: IdMinimalUser)
 
 // TODO: consider porting this to identity-play-auth.
 class AsyncAuthenticationService(
-  authenticationService: AuthenticationService,
-  testUserService: TestUserService
+    identityPlayAuthService: IdentityPlayAuthService,
+    testUserService: TestUserService
 )(implicit ec: ExecutionContext) {
 
-  // TODO in follow-up PR:
-  // authenticate the user by making a call to identity API rather than using underyling authentication service.
-  def authenticateUser(requestHeader: RequestHeader): Future[Option[AuthenticatedIdUser]] =
-    Future.successful(authenticationService.authenticatedUserFor(requestHeader))
+  import AsyncAuthenticationService._
+
+  def authenticateUser(requestHeader: RequestHeader): Future[AuthenticatedIdUser] =
+    identityPlayAuthService.getUserFromRequest(requestHeader)
+      .map { case (credentials, user) => buildAuthenticatedUser(credentials, user) }
+      .unsafeToFuture()
+
+  def tryAuthenticateUser(requestHeader: RequestHeader): Future[Option[AuthenticatedIdUser]] =
+    authenticateUser(requestHeader)
+      .map(user => Option(user))
```

Comment:
Option.apply handles `null` into `None`, is that something were expecting or is this better as 
```suggestion
      .map(user => Some(user))
```
or even
```suggestion
      .map(Some.apply)
```

Reply:
I think it's reasonable to treat `null` values as `None`, although I wouldn't expect a `null` to ever be returned.

Reply:
I agree with the first part, just when I see Option(blah) I always assume we're calling a java api that can return null and we're defending against that.
In second reading you're actually directly mapping both sides of a future to their respective sides of an option and removing the possibliity of a failed future.
Which might be better represented like replace the whole thing with this?
```
authenticateUser(requestHeader)
  .recover(failableUser => Success(failableUser.toOption))
```

Reply:
I don't think that would compile since `authenticateUser()` doesn't return an option (?) What we want is method takes two functions: `Throwable => B` and `A => B` but I was unable to find such a method 🤷‍♂️

Reply:
sorry I have no idea why I wrote recover, I meant `transform` which does exactly that as far as I remember

Reply:
Hmm not quite... `transform` either takes `Try[A] => Try[B]` or 2 functions: `Throwable => Throwable` and `A => B`, so I could do:
```scala
authenticateUser(requestHeader)
      .transform(user => Success(user.toOption))
```
but I'd probably have to introduce pattern matching to log the error so might end up being more verbose / less clear.

Reply:
oh yeah I forgot about the adding logging thing!  I think at this point we're probably discussing whether it's quicker to take the motorway around the town or take the back route through the side streets! 👍 interesting discussion though, thanks

Reply:
Ha, true! I do think it's interesting though that there isn't a method which does what we want; and it's not the only time I've had a need for something like this... Might raise with cats community and see what they say.